### PR TITLE
[WNMGDS-1924] RTL conversion of HC tests & enzyme removal

### PR DESCRIPTION
**Pull request by forrestbaer**
## Summary

- Updates all ds-healthcare-gov tests to use RTL
- Removes enzyme from the cmsds! 🎉 

## How to test

1. `yarn install && yarn test:unit`

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title`
- [x] Created or updated unit tests to cover the logic added
- [x] Updated unit test snapshots (`yarn update-snapshots`) and loki reference images **if necessary** (`yarn loki && yarn loki:healthcare && yarn loki:medicare`)

### CODE REVIEW

File: packages/ds-healthcare-gov/src/components/Header/ActionMenu.test.tsx [hunk 1]
```diff
@@ -89,15 +90,15 @@ describe('ActionMenu', function () {
...
+      expect(mock).toHaveBeenCalled();
     });
 
     it('sends analytics event when logged-out menu opened', () => {
       renderForAnalytics({
         links: [{ label: 'label', href: 'href' }],
       });
       fireEvent.click(screen.getByRole('button'));
-      expect(window.utag.link.mock.calls[0][0]).toMatchSnapshot();
+      expect(mock).toHaveBeenCalled();
```
Comment (pwolfert) [inline packages/ds-healthcare-gov/src/components/Header/ActionMenu.test.tsx:101]: Why were these snapshots removed?

**Review by pwolfert — COMMENTED** [review]
This is exciting!

I didn't look at all the snapshot changes for `packages/ds-healthcare-gov/src/components/Footer/__snapshots__/InlineLinkLists.test.tsx.snap`, which could be an indication that they're too verbose more than that I'm lazy.

One concern below about the analytics tests

File: packages/ds-healthcare-gov/src/components/Header/ActionMenu.test.tsx (hunk 1 above)
Comment (forrestbaer) [reply packages/ds-healthcare-gov/src/components/Header/ActionMenu.test.tsx:101]: oh good point I forgot about these, when I was working on them I had trouble finding these items within the rtl way of doing things.

Comment (forrestbaer) [reply packages/ds-healthcare-gov/src/components/Header/ActionMenu.test.tsx]: Ok @pwolfert these have been re-added

Comment (pwolfert) [reply packages/ds-healthcare-gov/src/components/Header/ActionMenu.test.tsx]: It looks like the snapshots that were added back in have all calls for all test instances, so they're really big and gnarly. The mock on `window.utag.link` doesn't reset between tests.

Comment (forrestbaer) [reply packages/ds-healthcare-gov/src/components/Header/ActionMenu.test.tsx:101]: Very true, took me a minute to figure this out, managed to grab only the needed data, updated!

Comment (forrestbaer) [reply packages/ds-healthcare-gov/src/components/Header/ActionMenu.test.tsx:101]: Whoops but now it's grabbing the wrong one.. looking into it.

Comment (forrestbaer) [reply packages/ds-healthcare-gov/src/components/Header/ActionMenu.test.tsx:103]: Ok actually fixed these, sorry for the noise.

**Review by pwolfert — APPROVED** [review]
LGTM

**Pull request merged**